### DISCUSSION TITLE
How to use a relative url in signIn() API?

### DISCUSSION THREAD

**Discussion by Layne101** [nextauthjs/next-auth#1874] [Help] [answered]
## Question 💬
The requirement is very simple here. I want to redirect to a relative URLs after sign in.
I noticed that the document says the signIn() API can support the relative URLs. I wonder if there is any example? how can I achieve that?

Attaching the document here.
Specifying a callbackUrl
The callbackUrl specifies to which URL the user will be redirected after signing in. It defaults to the current URL of a user.
You can specify a different callbackUrl by specifying it as the second argument of signIn(). This works for all providers.
The URL must be considered valid by the redirect callback handler. By default it requires the URL to be an absolute URL at the same hostname, or else it will redirect to the homepage. **You can define your own redirect callback to allow other URLs,** **including supporting relative URLs.**

**NOTE:** You can find doc [here](https://next-auth.js.org/getting-started/client#signin)!

**Comment by balazsorban44** [comment, marked as answer]
https://next-auth.js.org/configuration/callbacks#redirect-callback
